table/block_based_table_reader.h

```diff
@@ -528,11 +528,9 @@ class BlockBasedTableIterator : public InternalIterator {
     return data_block_iter_.value();
   }
   Status status() const override {
-    // It'd be nice if status() returned a const Status& instead of a Status
```

Comment:
Not particularly nice: `Status` is 16 bytes, so it's probably faster to return by value (in two registers).